### PR TITLE
Clarify that admin node should be upgraded first

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1073277
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
@tbazant @svenseeberg @Martin-Weiss

**Comment by Martin-Weiss** [comment]
FYI - had a look again and it seems that the whole process is having a wrong order for:

<SNIP>
Steps to Apply to All Cluster Nodes

1.    If you use SCC-registered systems, no actions have to be taken. Otherwise prepare software repositories. Disable all repositories except for SUSE Linux Enterprise Server 12 SP3 and SUSE Enterprise Storage 5 repositories. If these repositories are missing, add them. 
</SNIP>

If I do that for all servers in the cluster  they will all install salt packages from SLES 12 SP3 channels onto SLES 12 SP2 servers ! I assume that this is NOT supported at all.

I believe the correct process would be to install the salt-minions from the SLES 12 SP2 channels.. and attach the SLES 12 SP3 / SES 5 channels after that.

**Comment by smithfarm** [comment]
@Martin-Weiss Good point! I pushed a new version of the fix - please take another look.

File: xml/admin_ceph_upgrade.xml [hunk 1]
```diff
@@ -624,6 +624,11 @@ exported keyring for client.bootstrap-osd
      </para>
     </listitem>
    </itemizedlist>
+   <para>
+    In addition, before starting the procedure below, you need to upgrade the
+    &sadmin; node to &sls; 12 SP3 and &storage; 5 by running <command>zypper
```
Comment (svenseeberg) [inline xml/admin_ceph_upgrade.xml]: We have no sadmin entity. You can either add it to the entity-decl.ent or replace \&sadmin; with \&salt; master.

File: xml/admin_ceph_upgrade.xml [hunk 2]
```diff
@@ -689,6 +686,14 @@ exported keyring for client.bootstrap-osd
  "flags": "sortbitwise,recovery_deletes,purged_snapdirs",</screen>
     </tip>
    </step>
+   <step>
+    <para>
+     Upgrade the &sadmin; node to &sls; 12 SP3 and &storage; 5. If you use
+     SCC-registered systems, we recommend <command>zypper migration</command>.
+     After the upgrade, ensure that only repositories for &sls; 12 SP3 and
+     &storage; 5 are active (and refreshed) on the &sadmin; node before proceeding.
```
Comment (svenseeberg) [inline xml/admin_ceph_upgrade.xml]: We have no sadmin entity. You can either add it to the entity-decl.ent or replace \&sadmin; with \&salt; master.

File: xml/admin_ceph_upgrade.xml [hunk 3]
```diff
@@ -689,6 +686,14 @@ exported keyring for client.bootstrap-osd
  "flags": "sortbitwise,recovery_deletes,purged_snapdirs",</screen>
     </tip>
    </step>
+   <step>
+    <para>
+     Upgrade the &sadmin; node to &sls; 12 SP3 and &storage; 5. If you use
```
Comment (svenseeberg) [inline xml/admin_ceph_upgrade.xml]: We have no sadmin entity. You can either add it to the entity-decl.ent or replace \&sadmin; with \&salt; master.

**Comment by smithfarm** [comment]
@svenseeberg Done

@Martin-Weiss Please take another look.

**Comment by smithfarm** [comment]
@jschmid1 How does this look to you?

File: xml/admin_ceph_upgrade.xml [hunk 4]
```diff
@@ -624,6 +624,11 @@ exported keyring for client.bootstrap-osd
      </para>
     </listitem>
    </itemizedlist>
+   <para>
```
Comment (jschmid1) [inline xml/admin_ceph_upgrade.xml]: I think that'd tell the user to upgrade the admin _before_ setting up a salt cluster.
AFAIK the procedure starts with a SP2 based salt cluster, then upgrade the admin to SP3 in order to get the latest code which includes the engulfment

Comment (smithfarm) [reply xml/admin_ceph_upgrade.xml]: If you have a SP2-based salt cluster, and then upgrade the master node to SP3, that becomes a mixed Salt cluster (salt-master and salt-minion on the master node will be from SP3/SES5, and the rest of the minions will be SP2/SES4). That is the starting point for the engulfment and the maintenance.upgrade orchestration that follows.

**Comment by svenseeberg** [comment]
Feel free to merge this PR. I have no objections.

File: xml/admin_ceph_upgrade.xml (hunk 4 above)
Comment (jschmid1) [reply xml/admin_ceph_upgrade.xml]: ok, fine with me.

**Pull request merged**